Makefile

```diff
@@ -25,6 +25,7 @@ SHELL:=/usr/bin/env bash
 #
 GO_VERSION ?= 1.21.9
 GO_CONTAINER_IMAGE ?= docker.io/library/golang:$(GO_VERSION)
+GO_DIRECTIVE_VERSION ?= 1.21
```

Comment:
Shouldn't this have to be 1.21.0 (based on your comment in the script)

@chrischdi Do you remember when we had to specifcy the patch version as well? Was that something new with Go 1.22?

Reply:
Yeah, I think v1.22 introduced the "with patch" thing.

Reply:
I think upstream is now bumping to Go 1.22 across release branches. I think then we'll have Go 1.22 in CAPV also very soon

Reply:
So IIUC we will stay with allowing both 1.x (<=1.21) and 1.x.y (>1.22) formats?

Reply:
Maybe let's only support the latter

Reply:
You can test if it works with 1.21.0 as well. If it doesn't I would merge this PR right after we bumped to 1.22 (I can takeover the small rebase that might be necessary)

Reply:
Just saves us one iteration

Reply:
I think it's just a matter of days maybe 1-2 weeks before we bump

Reply:
No problem, I adjusted the regex to only include max versions for comparison in the format `1.x.y`. 
With an update to Go 1.22, only updating the max Go directive version [here](https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/pull/3004/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R28) should be fine.

Reply:
Thx!